plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1174,37 +1174,42 @@ public void testCheckpointWriteStatsAsStruct(String type, String inputValue, Str
                         tableName,
                         type,
                         getLocationForTable(bucketName, tableName)));
-        assertUpdate("INSERT INTO " + tableName + " SELECT " + inputValue, 1);
+        assertUpdate("INSERT INTO " + tableName + " SELECT " + sampleValue + " UNION ALL SELECT " + highValue, 2);
+
+        // TODO: Open checkpoint parquet file and verify 'stats_parsed' field directly
+        assertThat(getTableFiles(tableName))
+                .contains(getTableLocation(tableName) + "/_delta_log/_last_checkpoint");
 
         assertQuery(
                 "SHOW STATS FOR " + tableName,
                 "VALUES " +
-                        "('col', null, null, " + nullsFraction + ", null, " + statsValue + ", " + statsValue + ")," +
-                        "(null, null, null, null, 1.0, null, null)");
+                        "('col', null, null, " + nullsFraction + ", null, " + minValue + ", " + maxValue + ")," +
+                        "(null, null, null, null, 2.0, null, null)");
 
         assertUpdate("DROP TABLE " + tableName);
     }
 
     @DataProvider
     public Object[][] testCheckpointWriteStatsAsStructDataProvider()
     {
+        // type, sampleValue, highValue, nullsFraction, minValue, maxValue
         return new Object[][] {
-                {"boolean", "true", "0.0", "null"},
-                {"integer", "1", "0.0", "1"},
-                {"tinyint", "2", "0.0", "2"},
-                {"smallint", "3", "0.0", "3"},
-                {"bigint", "1000", "0.0", "1000"},
-                {"real", "0.1", "0.0", "0.1"},
-                {"double", "1.0", "0.0", "1.0"},
-                {"decimal(3,2)", "3.14", "0.0", "3.14"},
-                {"decimal(30,1)", "12345", "0.0", "12345.0"},
-                {"varchar", "'test'", "0.0", "null"},
-                {"varbinary", "X'65683F'", "0.0", "null"},
-                {"date", "date '2021-02-03'", "0.0", "'2021-02-03'"},
-                {"timestamp(3) with time zone", "timestamp '2001-08-22 03:04:05.321 -08:00'", "0.0", "'2001-08-22 11:04:05.321 UTC'"},
-                {"array(int)", "array[1]", "null", "null"},
-                {"map(varchar,int)", "map(array['foo', 'bar'], array[1, 2])", "null", "null"},
-                {"row(x bigint)", "cast(row(1) as row(x bigint))", "null", "null"},
+                {"boolean", "true", "false", "0.0", "null", "null"},
+                {"integer", "1", "2147483647", "0.0", "1", "2147483647"},
+                {"tinyint", "2", "127", "0.0", "2", "127"},
+                {"smallint", "3", "32767", "0.0", "3", "32767"},
+                {"bigint", "1000", "9223372036854775807", "0.0", "1000", "9223372036854775807"},
```

Comment:
Please add 

```
{"bigint", "1001", "2147483647", "0.0", "1001", "2147483647"},
```

in order to test also the branch used to test also the branch `if (jsonValue instanceof Integer) {` in `io.trino.plugin.deltalake.transactionlog.DeltaLakeParquetStatisticsUtils#jsonValueToTrinoValue`

Reply:
Both branches are already covered in the existing case.